### PR TITLE
feat: edit project namespace inline

### LINKED ISSUE #313: Update project: namespace disappears
When a project is modified, the namespace does not appear in the form any more until browser refresh (it is actually save correctly)
Dec-09-2020 13-53-04

### PR DESCRIPTION
**Pull request by martinmaillard**
Edit project namespace inline instead of in a nested form.

It should fix #313 and #329

### CODE REVIEW

File: ui/src/views/CubeProjectEdit.vue [hunk 1]
```diff
@@ -61,7 +62,8 @@ export default class CubeProjectEditView extends Vue {
       this.shape = await api.fetchOperationShape(operation)
     }
 
-    this.resource = Object.freeze(this.project?.pointer) ?? null
+    const namespace = this.project.csvMapping?.namespace ?? ''
+    this.resource = Object.freeze(this.project.pointer.addOut(cc.namespace, namespace))
```
Comment (tpluscode) [inline ui/src/views/CubeProjectEdit.vue:66]: I'd rather we could avoid this kind of tweaks.

**Review by tpluscode — COMMENTED** [review]
The validation can be improved by adding `sh:message` to the property shape of `cc:csvMapping`. Not perfect but enough

For the disappearing namespace, what is the actual cause?

File: ui/src/views/CubeProjectEdit.vue (hunk 1 above)
Comment (martinmaillard) [reply ui/src/views/CubeProjectEdit.vue:66]: I understand and I don't like it either, but having a nested form in the UI is not a good option either. There are multiple solutions to this problem:
- Have a separate screen to edit Mapping settings
- Magically provide `namespace` on the Project from the API
- Support complex sh:path in shaperone (e.g. `sh:path (cc:csvMapping cc:namespace)`)

For the disappearing namespace, I didn't analyze in detail, but I'm pretty sure it depends on if the full csvMapping is present in the dataset of the clownface pointer used in the form or not. In some cases, `project.csvMapping` is just a `link`. I think it worked by chance when first loading the form.

Comment (tpluscode) [reply ui/src/views/CubeProjectEdit.vue:66]: No good options. Fair, enough, let's have it like this for now

**Pull request merged**